### ISSUE TITLE
Is this project fading away?

### ISSUE THREAD

**Issue by crowdwave** [facebook/draft-js#1193]
I'm looking for a rich editor for reactjs but with 350 open issues and seemingly a trickle of commits I'm wondering if this project is still one to choose if I want one that is very actively developed and supported.

If not, can anyone suggest a more actively developed rich text solution for reactjs?

**Comment by robbertbrak** [comment]
That's a fair concern and one that I'm worried about myself. It seems that the original author is not around anymore and that there is very limited activity from the people at Facebook that have since taken over. Just take a look at the (lack of) 'weekly' meeting notes to get an idea: https://github.com/facebook/draft-js/tree/master/meta/meeting-notes.

If you are looking for a more actively developed Reactjs project, you could take a look at http://slatejs.org/. It has much better support for nested structures such as tables, so it may be more suitable for rich text editors that need complex functionality. Be aware, though, that Slatejs is not tested against IE/Edge and mobile devices, according to its FAQ (https://docs.slatejs.org/general/faq.html). Also, IME support seems terrible.

**Comment by flarnie** [comment]
This is absolutely a fair concern. There are still folks at Facebook actively maintaining this library, including myself, and we appreciate your patience since we also work on other open source projects. Draft is one of many options for a rich text editor, and it's worth looking at all the options and choosing what works best in your case.

Regarding the meeting notes - we didn't have much to report for the last few weeks so I have not been updating them on github, but in the future I will add some record of each meeting.

I'm about to do a minor release of the latest bug fixes and improvements - hopefully that is helpful to folks.

**Comment by robbertbrak** [comment]
Thanks for the reply, @flarnie! It's good to see in the meeting notes that you're also looking for ways to involve more maintainers both from Facebook and externally.

**Comment by crowdwave** [comment]
Of course no-one is obliged to work on this because its open source.

I ask because the community seems to say this is the flagship rich text editor for reactjs but my measures of a healthy active open source project - a low number of open issues plus a fairly constant flow of commits - suggest that this project is largely (but not entirely) abandoned.

**Comment by ironosity** [comment]
Over a year later, 480 open issues and 80 open PRs.  It's frustrating for my team because many of our pain points are open issues AND have open PRs...would be great to see Facebook put more resources at this project so at the very least, community PRs can be merged and more releases.

**Comment by flarnie** [comment]
Thanks for your comments - I set up a meeting with @sophiebits to come up with an update on the state of this project. You can expect that in the next few weeks.

**Comment by MathiasKandelborg** [comment]
I'm just leaving this here for anyone who likes alternatives to barely OS software: https://github.com/ianstormtaylor/slate

Also, please don't just develop open source software for your own needs FB. This could potentially be the best WYSIWYG editor out there, but they way you handle requests and don't want to do things you don't 'have the need for' is just. Idk. Don't pretend this is open source and at the same time keep all the good things for yourself.
I've spend so many hours trying to make DraftJS something and finally realised that you don't want to support the community, just yourself.

**Comment by RyanV** [comment]
is there any update @flarnie?

**Comment by flarnie** [comment]
@sophiebits and I did meet and discuss this, and while we have not yet come up with a solution that I'm satisfied with, we did agree that neither of us have time to maintain this project right now.

I can say that Draft.js is still widely used and essential across Facebook products, so we consider it very stable. We have set up automation to pull any internal fixes we make into the Github repository. I'll update this thread if and when there is any progress beyond that.

**Comment by thibaudcolas** [comment]
Thank you for the update @flarnie, it's good to know where Facebook / the project maintainers stand with regards to stability and ongoing maintenance.

Please let us know if there are things the wider community could do to help – I'm sure there are lots of people out there who would be keen to give a hand (e.g. like support on the [Draft.js Slack](https://draftjs.slack.com/)).

**Comment by ddelrio1986** [comment]
I would be willing to help out too and you can find me on the Draft.js Slack as well. So will you guys be accepting any bug fixes, enhancements, etc... from the community?

**Comment by tberman** [comment]
We (https://everyteam.co) rely on Draft, and maintain a fork that is up to date, has the 0.11 API, and we have added some useful things (like support for multiple entities per character).

I know given the current state, there hasn't been much interest in merging those changes back.

Does it make sense to create a separate branch inside this repo that can track FB upstream and allow the community to continue to push things forward or would FB (@flarnie, @sophiebits) prefer that happen outside of this repo?

**Comment by flarnie** [comment]
I have an update, some good news, and some context to share. Also want to start by thanking folks who continue to be involved and have shared feedback here.

**tldr:** Some new maintainers are starting, and we can expect a bit more activity! I'll let them make themselves known in other context - not sure this issue is the best place for an introduction. But of course they are welcome to chime in here. Since the repo now should be better maintained, I'll be closing this issue. :)

**Some more detailed context:**
*Why has nobody been looking at PRs/Issues?*

I am proud of the work I did cleaning up this repo in 2017 and earlier, but by the end of 2017 I lost opportunity to spend much time on Draft.js. The number of open issues and PRs spiked, and I waited in hopes that I could get more dedicated time for the project, or that a team at Facebook would pick up responsibility. But for now that has not happened, and I do regret letting the project stagnate while I was preoccupied. So I found some new folks to help! :)

Maintaining open source projects is often a "side project" opportunity for engineers at Facebook, when it makes sense for them and their manager. So the good news is there are 4 new maintainers spending some of their "side project" time to help clean up Draft.js.

**I'm very confident that this project is now actively maintained - 4 maintainers is much better than 1. :)**

*Why can't you merge my PR?*
Two things prevented your PR from landing -
- The high count of open PRs compared to time of maintainer to review them
- Low confidence of the reviewer that the change is safe and needed

Facebook basically only allows employees to merge PRs to open source projects, so fixing the first issue means increasing the time and number of open source maintainers. There are many of us internally fighting for this, and I'm continuing to look for ways that external folks can help with this.

For the second bullet point - we could use more tests, fixtures, and manual testing of PRs to gain confidence that they are safe. I would also advise contributors to focus on bug fixes and test coverage, as these are the things most likely to get landed.  And just be aware that it's never anything personal, we are all just doing the best we can. That's my best advice.
